testing/tests/DevExpress.ui.widgets.dataGrid/dataController.tests.js

```diff
@@ -3972,6 +3975,130 @@ QUnit.module('Virtual rendering', { beforeEach: setupVirtualRenderingModule, aft
         assert.deepEqual(dataSourceCall.args, [5], 'setViewportItemIndex call args');
     });
 
+    QUnit.test('addRow > scroll to far > scroll back', function(assert) {
+        // arrange
+        this.addRow();
+
+        // act
+        this.dataController.setViewportPosition(500);
+
+        // assert
+        assert.strictEqual(this.dataController.items()[0].key, 50, 'first item is new');
+
```

Comment:
Why is an item whose key value is 50 considered as 'new'?

Reply:
t's copy/paste. I will fix it.